examples/train/semantic_segmentation/main.py

```diff
@@ -0,0 +1,114 @@
+import hydra
+import numpy as np
+import torch
+import torch.nn as nn
+import torchvision
+from hydra.core.config_store import ConfigStore
+from hydra.utils import to_absolute_path
+from matplotlib.pyplot import figure
+
+import kornia as K
+from kornia.x import Configuration, Lambda, ModelCheckpoint, SemanticSegmentationTrainer
+
+cs = ConfigStore.instance()
+# Registering the Config class with the name 'config'.
+cs.store(name="config", node=Configuration)
+
+
+@hydra.main(config_path=".", config_name="config.yaml")
+def my_app(config: Configuration) -> None:
+
+    class Transform(nn.Module):
+        def __init__(self, image_size):
+            super().__init__()
+            self.resize = K.geometry.Resize(image_size, interpolation='nearest')
+
+        @torch.no_grad()
+        def forward(self, x, y):
+            x = K.utils.image_to_tensor(np.array(x))
+            x, y = x.float() / 255., torch.from_numpy(y)
+            return self.resize(x), self.resize(y)
+
+    # make image size homogeneous
+    transform = Transform(tuple(config.image_size))
+
+    # create the dataset
+    train_dataset = torchvision.datasets.SBDataset(
+        root=to_absolute_path(config.data_path), image_set='train', download=False, transforms=transform)
+
+    valid_dataset = torchvision.datasets.SBDataset(
+        root=to_absolute_path(config.data_path), image_set='val', download=False, transforms=transform)
+
+    # create the dataloaders
+    train_dataloader = torch.utils.data.DataLoader(
+        train_dataset, batch_size=config.batch_size, shuffle=True, num_workers=8, pin_memory=True)
+
+    valid_daloader = torch.utils.data.DataLoader(
+        valid_dataset, batch_size=config.batch_size, shuffle=True, num_workers=8, pin_memory=True)
+
```

Comment:
I am thinking if to build it towards a more less general purpose. The boiler code is too much, thinking of intergrating this part to the Trainer directly or so. So dose the preprocess and the postprocess.

Reply:
we can have a lighter layer later but this is plane pytorch in the end. Generalizing datasets is tricky since every problem/user usually have a totally random structure. Maybe a data structure that encapsulates train/valid dataloader - but in the next iteration once we increase the number of supported tasks and we figure out stable code patterns for what we support in terms of models. We don't want to be a general purpose thing